File: jmolecules-axon/02-axonstyle-cqrs-es-bank/src/main/java/org/jmolecules/example/axonframework/domain/api/command/atm/DepositMoneyCommand.java
package org.jmolecules.example.axonframework.domain.api.command.atm;

import org.jmolecules.architecture.cqrs.annotation.Command;
import org.jmolecules.ddd.annotation.Association;

@Command(namespace = "axon.bank", name = "DepositMoneyCommand")
public record DepositMoneyCommand(
  @Association
  String accountId,
  int amount
) {
  // empty
}

File: jmolecules-axon/03-onion-axon-cqrs-es-bank/src/main/kotlin/org/jmolecules/example/axonframework/domain/api/command/atm/DepositMoneyCommand.kt
package org.jmolecules.example.axonframework.domain.api.command.atm

import org.jmolecules.architecture.cqrs.annotation.Command
import org.jmolecules.ddd.annotation.Association
import org.jmolecules.example.axonframework.domain.api.type.AccountId
import org.jmolecules.example.axonframework.domain.api.type.Amount

/**
 * Deposit money command.
 */
@Command(namespace = "axon.bank", name = "DepositMoneyCommand")
data class DepositMoneyCommand(
    @Association
    val accountId: AccountId,
    val amount: Amount
)
